client/admin.go

```diff
@@ -59,31 +65,91 @@ func CreateAndInitTree(ctx context.Context, req *trillian.CreateTreeRequest, adm
 		return nil, err
 	}
 
-	if tree.TreeType == trillian.TreeType_MAP {
-		b.Reset()
-		err := b.Retry(ctx, func() error {
-			glog.Infof("Initialising Map %x...", tree.TreeId)
-			// Now, if it's a Map, initialise it.
-			req := &trillian.InitMapRequest{MapId: tree.TreeId}
-			resp, err := mapClient.InitMap(ctx, req)
-			if err != nil {
-				switch s, ok := status.FromError(err); {
-				case ok && s.Code() == codes.Unavailable:
-					glog.Errorf("Map server unavailable: %v", err)
-					return err
-				case ok && s.Code() == codes.AlreadyExists:
-					glog.Warningf("Bizarrely, the just-created Map (%x) is already initialised!: %v", tree.TreeId, err)
-					return err
-				}
-				return fmt.Errorf("failed to InitMap(%+v): %T %v", req, err, err)
+	switch tree.TreeType {
+	case trillian.TreeType_MAP:
+		err = InitMap(ctx, tree, mapClient)
+	case trillian.TreeType_LOG:
+		err = InitLog(ctx, tree, logClient)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return tree, nil
+}
+
+// InitMap initialises a freshly created Map tree.
+func InitMap(ctx context.Context, tree *trillian.Tree, mapClient trillian.TrillianMapClient) error {
+	if tree.TreeType != trillian.TreeType_MAP {
+		return fmt.Errorf("InitMap called with tree of type %v", tree.TreeType)
+	}
+
+	b := &backoff.Backoff{
+		Min:    100 * time.Millisecond,
+		Max:    10 * time.Second,
```

Comment:
Should max wait be configurable?

Reply:
Don't think it really matters, does it?
This is just the largest interval between retries rather than when it'll give up.

Reply:
Does it ever give up though?

Reply:
Yes, when the context is cancelled.

Reply:
I think m ax wait is badly named though I suspect it will turn out I named it.